pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -128,6 +125,13 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
 
+	// If the usersignup is deactivated then ensure the deactivated notification status is set to false, this is especially important for cases when a user is deactivated and then reactivated
+	if !instance.Spec.Deactivated {
+		if err := r.updateStatus(reqLogger, instance, r.setStatusDeactivationNotificationNotCreated); err != nil {
```

Comment:
Hm.. yeah. I didn't think about that but we do need to cover that case indeed. Good catch.
The only thing is that I would probably use a different reason. The CR is not created because it's not created doesn't sound super informative :)
I would use "UserIsActive" or "UserIsNotDeactivated" as the reason for the "NotificationCreated/False" condition :)

Reply:
Haha, yeah I guess it's not super informative. I changed it to `UserIsActive`

included in https://github.com/codeready-toolchain/host-operator/pull/289/commits/6bf0e589a480fce21f3bd1e4c69414ff0c73b23f